src/main/java/com/andersen/Ticket.java

```diff
@@ -0,0 +1,155 @@
+package com.andersen;
+
+import java.math.BigDecimal;
+
+public class Ticket implements Identifiable{
+
+	private static final int DISCOUNT = 2;
+
+	private int id;
+
+	@NullableWarning
+	private String concertHall;
+
+	@NullableWarning
+	private int eventCode;
+
+	private long timeStamp;
+
+	private boolean isPromo;
+
+	@NullableWarning
+	private char sector;
+
+	private double baggageLimit;
+
+	@NullableWarning
+	private BigDecimal price = BigDecimal.valueOf(20.50);
+
+	public Ticket() {
+		CheckNullFields.checkNullFields(this);
+	}
+
+	public Ticket(String concertHall, int eventCode, long timeStamp) {
+		this.concertHall = concertHall;
+		this.eventCode = eventCode;
+		this.timeStamp = timeStamp;
+	}
+
+	public Ticket(int id, String concertHall, int eventCode, long timeStamp, boolean isPromo, char sector, double baggageLimit) {
+		this.id = (id > 0 && id <= 9999) ? id : ++id;
+		if (concertHall.length() < 10) {
+			this.concertHall = concertHall;
+		} else {
+			this.concertHall = concertHall.substring(0, 10);
+		}
+		this.eventCode = (eventCode >= 100 && eventCode <= 999) ? eventCode : 100;
+		this.timeStamp = timeStamp;
+		this.isPromo = isPromo;
+		this.sector = sector;
+		this.baggageLimit = baggageLimit;
+		this.price = isPromo ? (price.subtract(BigDecimal.valueOf(DISCOUNT))) : price;
+	}
+
+	@Override
+	public int getId() {
+		return this.id;
+	}
+
+	@Override
+	public void setId(int id) {
+
+	}
+
+	public char getSector() {
+		return this.sector;
+	}
+
+	public void setSector(char sector) {
+		this.sector = sector;
+	}
+
+	public String getConcertHall() {
+		return concertHall;
+	}
```

Comment:
in sime metods you use this, in some not.. pls use the same (or **with** or without)